src/main/java/com/royalrangers/service/UserService.java

```diff
@@ -373,7 +380,7 @@ public void resendConfirmation(String email) throws UserRepositoryException, Unk
         if (!isEmailExist(email)) {
             throw new UserRepositoryException("User with this email is not exist.");
         }
-        if (user.getConfirmed()) {
+        if (!user.getConfirmed()) {
```

Comment:
Are you sure it should be !user.getCOnfirmed() here?
If yes - after user confirms it's email we should set *confirmed* onto false..